spec/02-integration/05-proxy/01-proxy_spec.lua

```diff
@@ -138,9 +139,49 @@ describe("#stream proxy interface listeners", function()
   end)
 end)
 
+
+local function reload_router(flavor)
+  _G.kong = {
+    configuration = {
+      router_flavor = flavor,
+    },
+  }
+
+  helpers.setenv("KONG_ROUTER_FLAVOR", flavor)
+
+  package.loaded["spec.helpers"] = nil
+  package.loaded["kong.global"] = nil
+  package.loaded["kong.cache"] = nil
+  package.loaded["kong.db"] = nil
+  package.loaded["kong.db.schema.entities.routes"] = nil
+  package.loaded["kong.db.schema.entities.routes_subschemas"] = nil
+
+  helpers = require "spec.helpers"
+
+  helpers.unsetenv("KONG_ROUTER_FLAVOR")
+end
+
+
+local function gen_route(flavor, r)
```

Comment:
Testing this way feels like we are making a player act judge at the same time.
Bascially we're only testing trad and `trad_compat`.

Reply:
We must use `gen_route()` to generate `expressions` field for the flavor `expressions`, or else we can not test it.